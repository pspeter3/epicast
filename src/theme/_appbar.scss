@import "./animation";
@import "./elevation";
@import "./sizing";
@import "./spacing";
@import "./typography";
@import "./variables";

$appbar__spacing: 8 * $unit;

.appbar {
    @include elevation-medium();
    background-color: $surface;
    color: $white;
    display: flex;
    justify-content: space-between;
    left: 0;
    padding: $subunit;
    position: fixed;
    right: 0;
    top: 0;
}

.appbar__navigation {
    @include sizing-row();
    display: flex;
}

.appbar__title {
    @include title();
    @include spacing-target();
    margin: 0;
}

.appbar__action {
    @include spacing-target();
    border-radius: 50%;
    color: $white;
    transition: background-color $duration-simple $easing-standard,
        color $duration-simple $easing-standard;

    &:focus,
    &:hover {
        background-color: $highlight;
        color: $primary;
    }
}

#root {
    padding: $appbar__spacing $subunit 0 * $unit;
}

@supports (padding: unquote("max(0px)")) {
    .appbar,
    #root {
        padding-left: unquote("max(#{$subunit}, env(safe-area-inset-left))");
        padding-right: unquote("max(#{$subunit}, env(safe-area-inset-right))");
    }

    .appbar {
        padding-top: unquote("max(#{$subunit}, env(safe-area-inset-top))");
    }

    #root {
        // SCSS cannot handle calc and env so we save it as a CSS variable
        --notch: env(safe-area-inset-top);
        padding-top: unquote("calc(#{$appbar__spacing} + var(--notch))");
        padding-bottom: unquote("env(safe-area-inset-bottom)");
    }
}
