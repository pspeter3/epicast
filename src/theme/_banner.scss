@import "./elevation";
@import "./spacing";
@import "./typography";
@import "./variables";

.banner {
    @include elevation-small();
    @include caption();
    @include spacing-y();
    background-color: $primary;
    border-radius: $border-radius;
    color: $white;
    text-align: center;
}
