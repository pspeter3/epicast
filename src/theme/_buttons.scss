@import "palette";
@import "spacing";
@import "typography";

.icon-button {
    @include target();
    display: block;
    color: $icon-inactive;

    &:hover,
    &:focus {
        color: $icon-active;
    }

    &:disabled {
        color: $icon-disabled;

        &:hover,
        &:focus {
            color: $icon-disabled;
        }
    }
}

@mixin text-button() {
    color: $text-hint;
    display: block;
    height: $leading-loose;
    text-align: center;

    &:hover,
    &:focus {
        color: $text-normal;
    }

    &:disabled {
        color: $text-disabled;

        &:hover,
        &:focus {
            color: $text-disabled;
        }
    }
}

.subheader-button {
    @include target-divider();
    @include text-button();
    @include typography-small();

    &:hover,
    &:focus,
    &[aria-pressed="true"] {
        @include target-border();
    }

    &[aria-pressed="true"] {
        border-bottom-color: $color-primary;
    }

    &:disabled {
        &:hover,
        &:focus {
            @include target-divider();
        }
    }

    &.numeric {
        text-align: center;
        width: 8 * $spacing-base;
    }

    &.primary {
        @include expand();
        text-align: center;
    }
}

.bottom-button {
    @include text-button();
    @include typography-small();
    border-top: $spacing-divider solid $color-divider;
    padding: ($spacing-small $spacing-divider) $spacing-base $spacing-small;
}
