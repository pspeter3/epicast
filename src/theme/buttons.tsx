import * as React from "react";

export const IconButton: React.SFC<React.ButtonHTMLAttributes<HTMLButtonElement>> = props => {
    return <button className="icon-button" type="button" {...props} />;
};

export const SubHeaderButton: React.SFC<React.ButtonHTMLAttributes<HTMLButtonElement>> = props => {
    return <button className="subheader-button" type="button" {...props} />;
};

export const NumericSubHeaderButton: React.SFC<
    React.ButtonHTMLAttributes<HTMLButtonElement>
> = props => {
    return <button className="subheader-button numeric" type="button" {...props} />;
};

export const PrimarySubHeaderButton: React.SFC<
    React.ButtonHTMLAttributes<HTMLButtonElement>
> = props => {
    return <button className="subheader-button primary" type="button" {...props} />;
};

export const BottomButton: React.SFC<React.ButtonHTMLAttributes<HTMLButtonElement>> = props => {
    return <button className="bottom-button" type="button" {...props} />;
};
