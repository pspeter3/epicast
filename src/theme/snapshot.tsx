import * as React from "react";
import TestRenderer from "react-test-renderer";

export const snapshot = (name: string, SFC: React.SFC<{}>): void => {
    describe(name, () => {
        it(`should match snapshot`, () => {
            const component = TestRenderer.create(<SFC>Snapshot</SFC>);
            const tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
};
