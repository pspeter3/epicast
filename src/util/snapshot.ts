import { shallow } from "enzyme";
import * as React from "react";

export type Factory = () => React.ReactElement<any>;

export const snapshot = (factory: Factory): (() => void) => {
    return () => {
        expect(shallow(factory())).toMatchSnapshot();
    };
};

export const snapshotSuite = (name: string, factory: Factory): void => {
    describe(name, () => {
        it("should match snapshot", snapshot(factory));
    });
};
