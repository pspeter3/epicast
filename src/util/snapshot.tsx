import * as React from "react";
import TestRenderer from "react-test-renderer";

export type Factory = () => React.ReactElement<any>;

export const snapshot = (factory: Factory): (() => void) => {
    return () => {
        const component = TestRenderer.create(factory());
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    };
};

export const snapshotSuite = (name: string, factory: Factory): void => {
    describe(name, () => {
        it("should match snapshot", snapshot(factory));
    });
};
