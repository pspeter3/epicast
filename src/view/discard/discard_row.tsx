import * as React from "react";
import { IconButton } from "../../theme/buttons";
import { TrashIcon } from "../../theme/icons";
import { Row } from "../../theme/layout";
import { Text } from "../../theme/typography";

export interface Props {
    city: string;
    onRemove: (city: string) => void;
}

export class DiscardRow extends React.PureComponent<Props, {}> {
    public render() {
        const { city } = this.props;
        return (
            <Row>
                <Text>{city}</Text>
                <IconButton onClick={this._onRemove}>
                    <TrashIcon />
                </IconButton>
            </Row>
        );
    }

    private _onRemove = () => {
        this.props.onRemove(this.props.city);
    };
}
