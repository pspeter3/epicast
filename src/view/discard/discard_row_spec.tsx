import { shallow } from "enzyme";
import * as React from "react";
import { IconButton } from "../../theme/buttons";
import { snapshot } from "../../util/snapshot";
import { DiscardRow } from "./discard_row";

describe("DiscardRow", () => {
    const createProps = () => {
        return {
            city: "San Francisco",
            onRemove: jest.fn(),
        };
    };

    it("should match snapshot", snapshot(() => <DiscardRow {...createProps()} />));

    it("should handle click", () => {
        const props = createProps();
        const tree = shallow(<DiscardRow {...props} />);
        tree.find(IconButton).simulate("click");
        expect(props.onRemove).toHaveBeenCalledWith(props.city);
    });
});
