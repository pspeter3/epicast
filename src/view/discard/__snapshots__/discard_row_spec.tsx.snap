// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscardRow should match snapshot 1`] = `
<Component>
  <Component>
    San Francisco
  </Component>
  <Component
    onClick={[Function]}
  >
    <Component />
  </Component>
</Component>
`;
